src/common/snippets/src/pass/mha_tokenization.cpp

```diff
@@ -95,26 +101,78 @@ auto tokenize_reshape_around_softmax(std::shared_ptr<ov::Node>& interm_op,
+                                       size_t& hidden_virtual_ports_count, size_t& potential_body_params_count) -> bool {
+    // TODO: Add Reshape support
+    while (is_supported_intermediate_op(interm_op)) {
         // All supported intermediate ops have only one output port
-        // To verify output element type is enough because all supported intermediate ops have the same output element type as input type
-        if (interm_op->get_output_target_inputs(0).size() != 1 || !is_supported_tensor(interm_op->get_output_tensor(0)))
+        if (interm_op->get_output_target_inputs(0).size() != 1)
             return false;
 
-        // Check for supported Broadcast op
+        // Check for supported ops on branches: Broadcast/Elementwise (for example, dequantize ops)
         if (interm_op->get_input_size() > 1) {
             tokenize_broadcast(interm_op, ordered_ops);
+
+            // To avoid unsupported number of non-scalar Constants in the future after FakeQuantize decomposition (plugin specific limitation)
+            // we should calculate potential number of non-scalar Constants for FakeQuantize that will be moved up from body.
+            if (const auto fq_node = ov::as_type_ptr<ov::op::v0::FakeQuantize>(interm_op)) {
+                hidden_virtual_ports_count += ngraph::snippets::utils::get_non_scalar_constant_count_for_fq(fq_node);
+            }
+
+            auto is_supported_branch_op = [&ordered_ops](const std::shared_ptr<ov::Node>& op) {
+                return is_supported_intermediate_op(op) &&
+                       ngraph::snippets::pass::GetSnippetsNodeType(op) != ngraph::snippets::pass::SnippetsNodeType::SkippedByPlugin &&
+                       std::find(ordered_ops.begin(), ordered_ops.end(), op) == ordered_ops.end();
+            };
+
+            for (size_t i = 0; i < interm_op->get_input_size(); ++i) {
+                const size_t shift = ordered_ops.size();
+                auto parent = interm_op->get_input_node_shared_ptr(i);
+                while (is_supported_branch_op(parent)) {
+                    // All supported ops have only one output port
+                    if (parent->get_output_target_inputs(0).size() != 1)
+                        break;
+
+                    // Add node only if there are scalar constants on inputs because of plugin-specific limitation
+                    bool are_weights_scalar = true;
+                    const auto parent_count = parent->get_input_size();
+                    for (size_t i = 1; i < parent_count; ++i) {
+                        are_weights_scalar = are_weights_scalar && ngraph::shape_size(parent->get_input_shape(i)) == 1;
+                    }
+
+                    ordered_ops.insert(ordered_ops.begin() + shift, parent);
+                    // We think that sequence of ops goes through input port 0
+                    // But can be Select here? If it can be, parent shouldn't be on input port 0. Need another way?
```

Comment:
Like with todos: let's state an assumption we are using here, and give a ticket id where the assumption will be analyzed and/or relaxed.

Reply:
Created the ticket and added number. Thanks

Reply:
Seems like the comment was not changed

Reply:
It's the result of rebasing. The comment is missed after that. Thank you for the catch! Fixed.